src/commands/moderateUsers/index.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ChatInputCommandInteraction,
+  SlashCommandBuilder,
+  GuildMember,
+  ThreadChannel,
+} from 'discord.js';
+import { Command } from '../command';
+import { isAdmin, isModerator } from '../../utils';
+
+const data = new SlashCommandBuilder()
+  .setName('removeuserbyrole')
+  .setDescription('Remove all users with a specific role from thread')
+  .addRoleOption((option) =>
+    option
+      .setName('name')
+      .setDescription('Tag a role to remove the list of matching users')
+      .setRequired(true)
+  );
+
+export const removeUserByRole = async (
+  interaction: ChatInputCommandInteraction
+) => {
+  const channel = interaction.channel as ThreadChannel;
+  if (!channel?.isThread) return;
```

Comment:
If not thread, interactions.reply("You can't remove all users with role from entire channel. This command only works in a thread.")

Reply:
Can you technically remove users from a channel?

Reply:
technically you can't, but this ensures it doesn't arrive at that "bad" path